test/functional/commands/touch-e2e-specs.js

```diff
@@ -0,0 +1,198 @@
+        {"type": "pointerDown", "button": 0},
+        {"type": "pointerMove", "duration": 200,  x: xEnd + 5, y: yEnd + 5},
+        {"type": "pointerUp", "button": 0}
+      ];
+      await performTouchAction(actions);
+      await driver.elementByXPath("//*[@text='ImageButton']");
+    });
+
+    it('should swipe up menu', async function () {
+      await (await driver.elementByAccessibilityId("Views")).click();
+      let gcEl = await driver.elementByAccessibilityId("Game Controller Input");
+      let {x, y} = await gcEl.getLocation();
+      let customEl = await driver.elementByAccessibilityId("Custom");
+      let {x:xEnd, y:yEnd} = await customEl.getLocation();
+
+      const actions = [
+        {"type": "pointerMove", "duration": 0,  x: x + 5, y: y + 5},
+        {"type": "pointerDown", "button": 0},
+        {"type": "pointerMove", "duration": 100,  x: xEnd + 5, y: yEnd + 5},
+        {"type": "pointerUp", "button": 0}
+      ];
+      await performTouchAction(actions);
+      await driver.elementByXPath("//*[@text='ImageButton']");
+    });
+  });
+
+  describe('touches', async function () {
+    it('should swipe on drag and drop', async function () {
+      await (await driver.elementByAccessibilityId("Views")).click();
+      await (await driver.elementByAccessibilityId("Drag and Drop")).click();
+      let el = await driver.elementById("io.appium.android.apis:id/drag_dot_1");
+      let {x, y} = await el.getLocation();
+      const touchActions = [
+        {"type": "pointerMove", "duration": 0, x: x + 30, y: y + 30},
+        {"type": "pointerDown", "button": 0},
+        {"type": "pointerMove", "duration": 100,  x: x + 10, y: y + 10},
+        {"type": "pointerUp", "button": 0}
+      ];
+      await performTouchAction(touchActions);
+    });
+
+    it('should touch down and up', async function () {
+      const el = await driver.elementByAccessibilityId("Accessibility");
+      let {x, y} = await el.getLocation();
+      const touchActions = [
+        {"type": "pointerMove", "duration": 0, x: x + 10, y: y + 10},
+        {"type": "pointerDown", "button": 0},
+        {"type": "pause", "duration": 100},
+        {"type": "pointerUp", "button": 0},
+      ];
+      await performTouchAction(touchActions);
+      await driver.elementByAccessibilityId("Accessibility Node Provider").should.eventually.exist;
+    });
+
+  });
+
+  describe('mjsonwp touch actions', function () {
+    it('should touch down at a location and then touch up', async function () {
+      let el = await driver.elementByAccessibilityId("Animation");
+      let sessionId = await driver.getSessionId();
+      let {x, y} = await el.getLocation();
+      const options = {
+        method: 'POST',
+        uri: `http://${HOST}:${PORT}/wd/hub/session/${sessionId}/touch/down`,
+        body: {
+          x: x + 1,
+          y: y + 1,
+        },
+        json: true,
+      };
+      await request(options);
```

Comment:
why call requests here rather than use driver methods?

Reply:
@jlipps wd doesn't have `/actions` driver methods yet.

Reply:
nvm... just noticed that it's `touch/down`

Reply:
I just checked again... `touch/down` isn't implemented in WD

Reply:
ah. well that would be a good thing to mention in a comment and add a todo to fix when it is in there